common/beerocks/bwl/dwpal/base_wlan_hal_dwpal.cpp

```diff
@@ -587,16 +587,22 @@ bool base_wlan_hal_dwpal::process_nl_events()
 }
 
 bool base_wlan_hal_dwpal::dwpal_nl_cmd_set(const std::string &ifname, unsigned int nl_cmd,
-                                           unsigned char *vendor_data, size_t vendor_data_size)
+                                           const unsigned char *vendor_data,
+                                           size_t vendor_data_size)
 {
     if (vendor_data == nullptr) {
         LOG(ERROR) << __func__ << "vendor_data is NULL ==> Abort!";
         return false;
     }
 
+    // Coping vendor_data to a local non-const variable in order
+    // to avoid passing a non-const reference in all upper layers.
+    unsigned char vendor_data_copy[vendor_data_size];
+    std::copy_n(vendor_data, vendor_data_size, vendor_data_copy);
```

Comment:
Why? Just to avoid a cast?
Doesn't make sense.
Also, why not change the `vendor_data` to `void*`?

Reply:
vendor_data is used in upper layer and scan functionality is depended on it .
so yes , i dont want to allow it to be changed in lower levels.